app/src/main/java/com/cornellappdev/android/eatery/MainActivity.java

```diff
@@ -64,9 +57,15 @@ protected void onCreate(Bundle savedInstanceState) {
         // Add functionality to bottom nav bar
 
         try {
-            BrbInfoModel model = (BrbInfoModel) InternalStorage.readObject(getApplicationContext(), CacheType.BRB);
-            Repository.getInstance().setBrbInfoModel(model);
-        } catch (IOException | ClassNotFoundException e) {
+            BrbInfoModel brbModel = (BrbInfoModel) InternalStorage.readObject(getApplicationContext(), CacheType.BRB);
+            Repository.getInstance().setBrbInfoModel(brbModel);
+            ArrayList<EateryBaseModel> campusEateries = (ArrayList<EateryBaseModel>) InternalStorage
+                    .readObject(getApplicationContext(), CacheType.CAMPUS_EATERY);
+            Repository.getInstance().setEateryList(campusEateries);
+            ArrayList<EateryBaseModel> ctownEateries = (ArrayList<EateryBaseModel>) InternalStorage
+                    .readObject(getApplicationContext(), CacheType.CTOWN_EATERY);
+            Repository.getInstance().setCtEateryList(ctownEateries);
+        } catch (IOException | ClassNotFoundException | ClassCastException e) {
```

Comment:
if we fail to get data in anyway, we should probably add some sort of pop-up or disable certain views here. otherwise, the app may crash during certain user flows if data is null.

Reply:
Yup - this happens when the app is first launched without access to WiFi. The app doesn't crash the app just displays a blank Eatery screen with no possible Eateries that can be clicked on. I 100% agree that this is something that should be looked at but I think it should be done in a separate PR - I don't think this PR exacerbates any issues with null data models.

Reply:
let's make a git issue for this as well. might need to talk to design about 'no wifi' prompts